snyk/policy
# feat: policy new ignore rules

- [ ] Reviewed by @remy (Snyk internal team)

#### What does this PR do?

- Disregards ignore rules if `disregardIfFixable` is set to `true` and the issue can be upgraded or patched
- Validates `reasonType` is one of the allowed reasonTypes ('not-vulnerable', 'wont-fix' or 'temporary-ignore')
- Validates `ignoredBy` is an object and that `ignoredBy.email` is an email

#### Where should the reviewer start?

Take a look at the tests.

#### How should this be manually tested?

You can test a `disregardIfFixable` gets applied by adding a package which has no upgrade or patch (e.g. `marked@0.3.5`) and adding another vulnerable module (e.g. `node-uuid@1.4.0`) which can be upgraded. Set both to be ignored with `disregardIfFixable` set to true, and only one should be ignored.
